Examples/Cocoa/Tests/Objective_CTests/ObjcTests.swift

```diff
@@ -305,9 +305,58 @@ class ObjcDictionaryRepresentationTestSuite: XCTestCase {
         assertDictionaryRepresentation(dict)
     }
 
+    func testCharEnum() {
+        let dict: JSONDict = [
+            "char_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testUnsignedCharEnum() {
+        let dict: JSONDict = [
+            "unsigned_char_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testShortEnum() {
+        let dict: JSONDict = [
+            "short_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testUnsignedShortEnum() {
+        let dict: JSONDict = [
+            "unsigned_short_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
     func testIntEnum() {
         let dict: JSONDict = [
-            "int_enum": 2,
+            "int_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testUnsignedIntEnum() {
+        let dict: JSONDict = [
+            "unsigned_int_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testNSIntegerEnum() {
+        let dict: JSONDict = [
+            "nsinteger_enum": 0,
+        ]
+        assertDictionaryRepresentation(dict)
+    }
+
+    func testNSUIntegerEnum() {
+        let dict: JSONDict = [
+            "nsuinteger_enum": 0,
```

Comment:
Can we add more tests for the other capabilities of models like merging, serialization, etc to ensure they are working for these changes to enum types?